mdata/cache/ccache.go

```diff
@@ -94,29 +122,47 @@ func (c *CCache) CacheIfHot(metric string, prev uint32, itergen chunk.IterGen) {
 	accnt.CacheChunkPushHot.Inc()
 
 	c.RUnlock()
-	c.Add(metric, prev, itergen)
+	met.Add(prev, itergen)
 }
 
-func (c *CCache) Add(metric string, prev uint32, itergen chunk.IterGen) {
+func (c *CCache) Add(metric, rawMetric string, prev uint32, itergen chunk.IterGen) {
 	c.Lock()
 	defer c.Unlock()
 
-	if ccm, ok := c.metricCache[metric]; !ok {
+	ccm, ok := c.metricCache[metric]
+	if !ok {
 		ccm = NewCCacheMetric()
-		ccm.Init(prev, itergen)
+		ccm.Init(rawMetric, prev, itergen)
 		c.metricCache[metric] = ccm
+
+		// if we do not have this raw key yet, create an entry for it
+		ccms, ok := c.metricRawKeys[rawMetric]
+		if !ok {
+			ccms = make(map[string]struct{})
+			c.metricRawKeys[rawMetric] = ccms
+		}
+
+		// if we don't have it yet, associate the metric with this raw key
+		if _, ok = ccms[metric]; !ok {
+			ccms[metric] = struct{}{}
```

Comment:
note if we're adding the raw one itself, we don't need to associate it with itself (e.g. `if metric != rawMetric`)

Reply:
I think that would make `Reset()` and `DelMetric()` unnecessarily complicated. 
Also, it would actually make it impossible to return the accurate series count of deleted metrics from `Reset()`, because if the `rawMetric` is not in `metricRawKeys` (f.e. if only the raw metric has been added without any archive) then reset can't reliably know how many series have been cleared, because inside `.metricCache` the only way to determine whether a key is a raw metric key or an archive key would be by pattern matching for the archive suffix like `_sum_<ttl>` (but that would be really ugly).

Reply:
> if the rawMetric is not in metricRawKeys (f.e. if only the raw metric has been added without any archive) then reset can't reliably know how many series have been cleared,

isn't it just a matter of following the associations? if there's 1 key to delete and it has no associations, then it's just that key, no?